test/common_test/basic_test.cc

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#include <gtest/gtest.h>
+
+#include <gz/common/Filesystem.hh>
+#include <gz/common/Console.hh>
+#include <gz/common/Util.hh>
+#include <gz/plugin/Loader.hh>
+
+#include <gz/physics/GetEntities.hh>
+#include <gz/physics/RequestEngine.hh>
+
+class EntityManagementFeaturesTest:
+  public testing::Test,
+  public testing::WithParamInterface<const char *>
+{
+  // Documentation inherited
+  public: void SetUp() override
+  {
+    gz::common::Console::SetVerbosity(4);
+
+    if (!gz::common::env("PHYSICS_ENGINE_NAME", physicsEngineName))
```

Comment:
I think we shouldn't need to specify `PHYSICS_ENGINE_NAME`. If it's not specified, we should use `FindFeatures3d` to identify the plugin names contained within the `LIB_TO_TEST` and then test each of those plugins. To support this, we should get the list of pluginNames in `SetUp` and move the `gz::plugin::Loader` instantiation and `LoadLib` call to a loop in the test case

* https://github.com/gazebosim/gz-physics/blob/e91c58a49522cfdc5793fe8b980566eb26531ba4/examples/hello_world_loader/hello_world_loader.cc#L51-L63

Reply:
https://github.com/gazebosim/gz-physics/pull/357/commits/1f48f3aead7cfea6948fb4d980e7c3f840badadb